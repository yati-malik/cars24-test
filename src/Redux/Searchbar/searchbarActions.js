import { SET_SEARCH } from "../Common/actionTypes";

export function SetSearch(name) {
  return {
    type: SET_SEARCH,
    name: name
  };
}
