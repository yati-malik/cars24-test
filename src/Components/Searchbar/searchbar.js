import React, { useState, forwardRef, useImperativeHandle, useEffect } from "react";
import { ToastContainer, toast } from "react-toastify";
import {SetSearch as setSearch} from "../../Redux/Searchbar/searchbarActions";
import { connect } from "react-redux";

const Searchbar = (props) => {
  let [product, setProduct] = useState(props.name);

  useEffect(() => {
    setProduct(props.name);
  },[props.name])

  let handleOnChange = (e) => {
    setProduct(e.target.value);
  };
  let handleProductSearch = () => {
    if (product == "") {
      toast.warn("Search field value is empty !", {
        position: toast.POSITION.TOP_RIGHT,
      });
    } else {
      props.setSearch(product);
    }
  };

  return (
    <div className="search-bar">
      <ToastContainer autoClose={3000} />
      <div className="search-container">
        <input
          type="text"
          value={product}
          placeholder="Name"
          className="search-box"
          onChange={handleOnChange}
        />
        <input
          type="button"
          className="search"
          value="Search"
          onClick={handleProductSearch}
        />
      </div>
    </div>
  );
};

const mapStateToProps = (state, ownProps) => {
  const { name } = state.productReducer;
  return { name };
};

export default connect(mapStateToProps, { setSearch })(Searchbar);
