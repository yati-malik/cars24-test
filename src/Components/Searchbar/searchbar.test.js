import React from "react";
import { configure, mount, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Searchbar from "./searchbar";
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { render } from "@testing-library/react";

const mockStore = configureStore();
const productReducer = {
    name: '',
    filters: {
        startPrice: 0,
        endPrice: 0
    }
}

configure({ adapter: new Adapter() });

describe('<Searchbar />',() => {
    let searchbar;
    let store;

    beforeEach(() =>{
        store = mockStore({productReducer});
        searchbar = renderer.create(
            <Provider store = {store}>
                <Searchbar />
            </Provider>
        );
    })

    it('renders correctly', () => {
        expect(searchbar.toJSON()).toMatchSnapshot();
    })

});