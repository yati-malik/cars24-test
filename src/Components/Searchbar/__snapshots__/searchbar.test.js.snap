// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Searchbar /> renders correctly 1`] = `
<div
  className="search-bar"
>
  <div
    className="Toastify"
  />
  <div>
    <input
      className="form-control"
      onChange={[Function]}
      placeholder="Name"
      type="text"
      value=""
    />
    <input
      className="search"
      onClick={[Function]}
      type="button"
      value="Search"
    />
  </div>
</div>
`;
